prow/cmd/deck/static/common/rerun.ts

```diff
@@ -0,0 +1,151 @@
+import {alertToast, copyToClipboardWithToast, icon, setToast} from "./common";
+import {relativeURL} from "./urls";
+
+export function createRerunProwJobIcon(modal: HTMLElement, rerunElement: HTMLElement, prowjob: string, rerunCreatesJob: boolean, csrfToken: string): HTMLElement {
+  const LATEST = 'latest';
+  const ORIGINAL = 'original';
+  const inrepoconfigURL = 'https://github.com/kubernetes/test-infra/blob/master/prow/inrepoconfig.md';
+  const latestURL = `${location.protocol}//${location.host}/rerun?mode=latest&prowjob=${prowjob}`;
+  const originalURL = `${location.protocol}//${location.host}/rerun?mode=original&prowjob=${prowjob}`;
+  let commandURL = latestURL;
+  const i = icon.create("refresh", "Show instructions for rerunning this job");
+
+  window.onkeydown = (event: any) => {
+    if ( event.key === "Escape" ) {
+      modal.style.display = "none";
+      // Resets modal content. If removed elements will be concatenated, causing duplicates.
+      rerunElement.innerHTML = '';
+    }
+  };
+  window.onclick = (event: any) => {
+    if (event.target === modal) {
+      modal.style.display = "none";
+      // Resets modal content. If removed elements will be concatenated, causing duplicates.
+      rerunElement.innerHTML = '';
+    }
+  };
+  
+  const getCommandDescription = (mode: string): string => {
+    return `The command is for the ${mode} configuration. Admin permissions are required to rerun via kubectl.`
+  }
+  const getCommand = (url: string):string => {
+    return `kubectl create -f "${url}"`
+  };
+
+  // we actually want to know whether the "access-token-session" cookie exists, but we can't always
+  // access it from the frontend. "github_login" should be set whenever "access-token-session" is
+  i.onclick = () => {
+    modal.style.display = "block";
+
+    rerunElement.innerHTML = `
+      <h2 class="rerunModal-title">Rerun ProwJob</h2>
+      <p class="rerunModal-description">
+        Below you can choose to rerun this ProwJob with either the original or latest configuration. 
+        <br><br> 
+        Note: Rerunning a ProwJob will create a new instance of a ProwJob. Any ProwJobs already running will not be interrupted.
+        <a href="${inrepoconfigURL}" target="_blank">
+          Inrepoconfig
+          <i class='rerunModal-openOut material-icons state triggered' style='color: gray'>open_in_new</i>
```

Comment:
I like to see double-quotes in HTML (and single-quotes in JS). Per previous/personal code conventions. But both quotation types are valid in both contexts. (Since this is actual/literal HTML that _just happens_ to be composed in JS, my brain is expecting double-quotes.)

Reply:
Agreed, updated!